functions/snippets/src/main/java/com/example/functions/FirebaseFirestoreReactive.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.functions;
+
+// [START functions_firebase_reactive]
+import com.google.api.client.json.Json;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.cloud.firestore.SetOptions;
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.cloud.functions.RawBackgroundFunction;
+import com.google.cloud.logging.LoggingHandler;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+import org.apache.commons.lang3.concurrent.ConcurrentException;
+
+public class FirebaseFirestoreReactive implements RawBackgroundFunction {
+
+  // Use GSON (https://github.com/google/gson) to parse JSON content.
+  private Gson gsonParser = new Gson();
+
+  private static final Logger LOGGER = Logger.getLogger(FirebaseFirestoreReactive.class.getName());
+  private static final Firestore firestore = FirestoreOptions.getDefaultInstance().getService();
+
+  @Override
+  public void accept(String json, Context context) throws RuntimeException {
+    // Get the recently-written value
+    JsonObject body = gsonParser.fromJson(json, JsonObject.class);
+    JsonObject fields = body.getAsJsonObject("value").getAsJsonObject("fields");
+    if (!fields.has("original")) {
+      return;
+    }
+
+    // Convert recently-written value to ALL CAPS
+    String currentValue = fields.getAsJsonObject("original").get("stringValue").getAsString();
+    String newValue = currentValue.toUpperCase();
+
+    // Update Firestore DB with ALL CAPS value
+    Map<String, String> newFields = new HashMap<>();
+    newFields.put("original", newValue);
+
+    String affectedDoc = context.resource().split("/documents/")[1].replace("\"", "");
+
+    LOGGER.info(String.format("Replacing value: %s --> %s", currentValue, newValue));
+    try {
+      firestore.document(affectedDoc).set(newFields, SetOptions.merge()).get();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
Why do you bother with this?  Don't bother catching or even the try if you aren't going to do something useful with it.  It will automatically propagate to the parent.

Reply:
This throws `InterruptedException` and `ConcurrentException`, but this `accept` method can only throw `RuntimeException`.

I'll change this handler to be more specific, though (and only catch the two exception types I mentioned above).